core/templates/pages/exploration-editor-page/editor-tab/templates/modal-templates/add-answer-group-modal.component.html

```diff
@@ -31,47 +35,47 @@ <h3>Add Response</h3>
     </div>
     <br>
 
     <oppia-outcome-destination-editor [outcome]="tmpOutcome"
-                                      (get-changes)="getChanges()"
-                                      ng-if="!questionModeEnabled"
-                                      [outcome-has-feedback]="!isLinearWithNoFeedback(tmpOutcome)"
-                                      (add-state)="addState($event)">
+                                      *ngIf="!questionModeEnabled"
+                                      [outcomeHasFeedback]="true"
+                                      (addState)="updateState($event)">
     </oppia-outcome-destination-editor>
   </form>
 
-  <div class="oppia-correctness-label-editor" ng-if="isCorrectnessFeedbackEnabled() && !isCurrentInteractionLinear()">
+  <div class="oppia-correctness-label-editor" *ngIf="isCorrectnessFeedbackEnabled() && !isCurrentInteractionLinear()">
     <strong>The answers in this group are correct</strong>
-    <input class="protractor-test-editor-correctness-toggle" type="checkbox" ng-model="tmpOutcome.labelledAsCorrect">
+    <input class="protractor-test-editor-correctness-toggle" type="checkbox" [(ngModel)]="tmpOutcome.labelledAsCorrect">
   </div>
-  <oppia-question-misconception-editor ng-if="questionModeEnabled"
-                                       (save-tagged-misconception)="updateTaggedMisconception($event)"
-                                       (save-answer-group-feedback)="updateAnswerGroupFeedback($event)"
-                                       [is-editable]="true"
+  <oppia-question-misconception-editor *ngIf="questionModeEnabled"
+                                       (saveTaggedMisconception)="updateTaggedMisconception($event)"
+                                       (saveAnswerGroupFeedback)="updateAnswerGroupFeedback($event)"
+                                       [isEditable]="true"
                                        [outcome]="tmpOutcome"
                                        [rules]="tmpRule"
-                                       [tagged-skill-misconception-id]="tmpTaggedSkillMisconceptionId">
+                                       [taggedSkillMisconceptionId]="tmpTaggedSkillMisconceptionId">
   </oppia-question-misconception-editor>
 </div>
 
 <div class="modal-footer">
-  <button class="btn btn-secondary protractor-test-close-add-response-modal" ng-click="cancel()">Cancel</button>
-  <button class="btn btn-success protractor-test-add-new-response" ng-click="saveResponse(false)" ng-disabled="(isInvalid || addAnswerGroupForm.outcomeDetailsForm.$invalid) || isSelfLoopWithNoFeedback(tmpOutcome) || isFeedbackLengthExceeded(tmpOutcome)">Save Response</button>
-  <button class="btn btn-success" ng-click="saveResponse(true)" ng-disabled="(isInvalid || addAnswerGroupForm.outcomeDetailsForm.$invalid) || isSelfLoopWithNoFeedback(tmpOutcome) || isFeedbackLengthExceeded(tmpOutcome)">Save and Add Another</button>
```

Comment:
I see that `addAnswerGroupForm.outcomeDetailsForm.$invalid` is removed, is there an equivalent validation in place?

Reply:
https://user-images.githubusercontent.com/77006897/177302007-84c5bfa9-e62f-442a-b4a2-9b94a5973e4e.mp4

`we are not using this feature earlier we are not using this feature now dead code`

Reply:
Hi @heyimShivam, could you please explain how you verified this "we are not using this feature earlier"?

Also, in the video, I see a scrollbar of 0 length (right side of "Add element" button), is that expected?

Reply:
yes, this is already there in code Schema-based-editor

Reply:
Oh, I see. Please could you create an issue on Github to track this as it does not look great.